### PR TITLE
fix: some vulnerabilities

### PR DESCRIPTION
**Pull request by Florian-Schoenherr**
## Purpose of this pull request? 

- [ ] Documentation update
- [X] Bug fix 
- [ ] Enhancement
- [ ] Other, please explain:

## What changes did you make?

Fixed some of the security vulnerabilities, the others depend on some of the PRs in sub repos.

## Is there anything you'd like reviewers to focus on?

I removed the "pretest": "xo" script for now, as it doesn't work. I think temporarily removing it is better than keeping a high security vulnerability around. This is the error when keeping it:
image
the cut line says `Cannot find package 'D:\...\yo\node_modules\xo\node_modules\eslint\' imported from D:\...\yo\node_modules\xo\index.js`

### CODE REVIEW

**Comment by Florian-Schoenherr** [comment]
oh and I replaced got with node-fetch, which wasn't really necessary here, could be reverted, too.
Although does the fetch right there even work? If I just go on that link it gives back a 404 page.

File: package.json [hunk 1]
```diff
@@ -17,7 +17,6 @@
   "scripts": {
     "postinstall": "yodoctor",
     "postupdate": "yodoctor",
-    "pretest": "xo",
```
Comment (mshima) [inline package.json:20]: We should keep pretest script.

File: package.json [hunk 2]
```diff
@@ -84,7 +82,7 @@
     "proxyquire": "^2.0.1",
     "registry-url": "^5.1.0",
     "sinon": "^6.1.3",
-    "xo": "0.38.0"
+    "xo": "^0.47.0"
```
Comment (mshima) [inline package.json:85]:
```suggestion
    "xo": "0.38.0"
```
There are security problems, but this is dev only, won’t be a problem for npm install yo.
Version 0.40.0 is esm only. So 0.39.x should be to newest possible.

**Review by mshima — CHANGES_REQUESTED** [review]

**Comment by UlisesGascon** [comment]
It looks like this has been inactive for some time, so we're going to go ahead and close it for now to keep the repo tidy.

If you'd like to revisit this or have any updates to share, please feel free to reopen the PR or create a new one—your contributions are always welcome @Florian-Schoenherr!

**Pull request closed without merging**